tsdb/head.go

```diff
@@ -1321,6 +1322,8 @@ func (h *Head) gc() {
 	symbols := make(map[string]struct{}, len(h.symbols))
 	values := make(map[string]stringset, len(h.values))
 
+	h.symMtx.Lock()
```

Comment:
Another question: in `getOrCreateWithID()` we do wrap `h.postings.Add()` with the lock, but in this function we do call `h.postings.Delete()` outside of the lock. Could this introduce another race?

Reply:
Answering to myself: no, it shouldn't, because what we need to protect is a de-sync between postings and symbols. The mere call to `h.postings.Delete()` doesn't touch symbols. It's in the `h.postings.Iter()` that we do build the map of updated symbols, so that's what we have to protect.

Reply:
yup :+1: